hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaDeploymentManagement.java

```diff
@@ -430,72 +423,71 @@ public Action forceQuitAction(final Long actionId) {
+            final Long rolloutGroupParentId, final int limit) {
         final DefaultTransactionDefinition def = new DefaultTransactionDefinition();
         def.setName("startScheduledActions");
         def.setReadOnly(false);
         def.setIsolationLevel(Isolation.READ_UNCOMMITTED.value());
         def.setPropagationBehavior(TransactionDefinition.PROPAGATION_REQUIRES_NEW);
         return new TransactionTemplate(txManager, def).execute(status -> {
-            final Page<Action> rolloutGroupActions = findActionsByRolloutAndRolloutGroupParent(rollout,
-                    rolloutGroupParent, limit);
+            final Page<Action> rolloutGroupActions = findActionsByRolloutAndRolloutGroupParent(rolloutId,
+                    rolloutGroupParentId, limit);
 
             rolloutGroupActions.map(action -> (JpaAction) action).forEach(this::startScheduledAction);
 
             return rolloutGroupActions.getTotalElements();
         });
     }
 
-    private Page<Action> findActionsByRolloutAndRolloutGroupParent(final Rollout rollout,
-            final RolloutGroup rolloutGroupParent, final int limit) {
-        final JpaRollout jpaRollout = (JpaRollout) rollout;
-        final JpaRolloutGroup jpaRolloutGroup = (JpaRolloutGroup) rolloutGroupParent;
+    private Page<Action> findActionsByRolloutAndRolloutGroupParent(final Long rolloutId,
+            final Long rolloutGroupParentId, final int limit) {
+
         final PageRequest pageRequest = new PageRequest(0, limit);
-        if (rolloutGroupParent == null) {
-            return actionRepository.findByRolloutAndRolloutGroupParentIsNullAndStatus(pageRequest, jpaRollout,
+        if (rolloutGroupParentId == null) {
+            return actionRepository.findByRolloutIdAndRolloutGroupParentIsNullAndStatus(pageRequest, rolloutId,
                     Action.Status.SCHEDULED);
         } else {
-            return actionRepository.findByRolloutAndRolloutGroupParentAndStatus(pageRequest, jpaRollout,
-                    jpaRolloutGroup, Action.Status.SCHEDULED);
+            return actionRepository.findByRolloutIdAndRolloutGroupParentIdAndStatus(pageRequest, rolloutId,
+                    rolloutGroupParentId, Action.Status.SCHEDULED);
         }
     }
 
-    private Action startScheduledAction(final JpaAction action) {
+    private void startScheduledAction(final JpaAction action) {
         // check if we need to override running update actions
         final Set<Long> overrideObsoleteUpdateActions = overrideObsoleteUpdateActions(
                 Collections.singletonList(action.getTarget().getId()));
 
-        if (action.getTarget().getAssignedDistributionSet() != null && action.getDistributionSet().getId()
-                .equals(action.getTarget().getAssignedDistributionSet().getId())) {
+        if (((JpaTarget) action.getTarget()).getAssignedDistributionSet() != null && action.getDistributionSet().getId()
```

Comment:
For better reagability use a variable for the cast `(JpaTarget) action.getTarget()`. Then you have only 1 instead of 2 cast